### PR TITLE
feat: Add testing documentation

### PR DESCRIPTION
**Pull request by lievenvg**
## PR Checklist

This PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)
- [ ] A changelog entry has been added to CHANGELOG.md if necessary

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] react application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #123 

## What is the new behavior?
Update for testing documentation.
And made a small `<i>` to `<span>` correction on the Icon component.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Resolved issues

Resolves: #123

### CODE REVIEW

File: TESTING.md [hunk 1]
```diff
@@ -1,6 +1,153 @@
...
+# ACPaaS UI Testing Guide
 
-To make life easier in VScode for testing with Jest.
-We suggest you install the following plugin:
+When developing, testing is a major part of the work.
+We use [Jest](https://jestjs.io) in combination with [Enzyme](https://airbnb.io/enzyme) as our test suite.
 
-https://github.com/jest-community/vscode-jest
\ No newline at end of file
+## VSCode testing
```
Comment (TriangleJuice) [inline TESTING.md]: I'm not sure if this section is as important as the next one. I would suggest to move this info to the bottom or to use no heading.

File: TESTING.md [hunk 2]
```diff
@@ -1,6 +1,153 @@
-# VSCode testing
+# ACPaaS UI Testing Guide
 
-To make life easier in VScode for testing with Jest.
-We suggest you install the following plugin:
+When developing, testing is a major part of the work.
```
Comment (TriangleJuice) [inline TESTING.md]: Is it? 😏

File: TESTING.md [hunk 3]
```diff
@@ -1,6 +1,153 @@
...
+To make life easier in Visual Studio Code for testing with Jest, we suggest that you install [this VS Code Jest extension](https://github.com/jest-community/vscode-jest).
+
+## ACPaaS UI Test Automation Guide
+
+When automating tests, one of the challenges is to keep the test resilient to code changes.
+Using [data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) can help you with that, by creating unique selectors for your tests.
+
+### Using unique data attributes with ACPaaS UI
+
+It's not possible to add unique data attributes to the ACPaaS UI component library for the simple reason that the attribute won't be unique anymore when you use the component multiple times on the same page.
```
Comment (TriangleJuice) [inline TESTING.md]:
```suggestion
It's not possible to add unique attributes to the ACPaaS UI components for the simple reason that the attribute won't be unique anymore when you use the component multiple times on the same page.
```

File: TESTING.md [hunk 4]
```diff
@@ -1,6 +1,153 @@
...
+
+But in your app you can simply add a data attribute to each individual ACPaaS UI component so you can target the component that you want to test.
```
Comment (TriangleJuice) [inline TESTING.md]:
```suggestion
We solve this problem by adding unique attributes to the individual ACPaaS UI components at app level.
```

File: TESTING.md [hunk 5]
```diff
@@ -1,6 +1,153 @@
...
+
+Which will result in:
+
+```html
+<div class="m-accordion" data-qa="id-1234">
+  ...
+</div>
+```
+
+### Branding classes
```
Comment (TriangleJuice) [inline TESTING.md]:
```suggestion
### Targeting nested elements
```

File: TESTING.md [hunk 6]
```diff
@@ -1,6 +1,153 @@
...
+
+Since you probably want to target a specific element or a subcomponent, selecting the parent component won't be enough.
```
Comment (TriangleJuice) [inline TESTING.md]:
```suggestion
If you want to target a specific element within the component, use their CSS class in combination with the unique data attribute of the parent component.
```

File: TESTING.md [hunk 7]
```diff
@@ -1,6 +1,153 @@
...
+You can however target these elements and subcomponents based on their CSS classes in combination with the unique data attribute of the parent component.
```
Comment (TriangleJuice) [inline TESTING.md]:
```suggestion

```

File: TESTING.md [hunk 8]
```diff
@@ -1,6 +1,153 @@
...
+
+#### Example
+
+In this example the accordiontab can be opened or closed by toggling the `<div>` tag with the CSS class `m-accordion__tab`, which is nested in the accordion with data attribute `data-qa="id-1234"`.
```
Comment (TriangleJuice) [inline TESTING.md]:
```suggestion
In the following example the accordion tab can be opened or closed by toggling the `<div>` tag with the CSS class `m-accordion__tab`, which is nested in the accordion with data attribute `data-qa="id-1234"`.
```

File: TESTING.md [hunk 9]
```diff
@@ -1,6 +1,153 @@
...
+
+```html
+<div class="m-accordion" data-qa="id-1234">
+  <div ... class="m-accordion__tab">
+  ...
+</div>
+```
```
Comment (TriangleJuice) [inline TESTING.md]: Maybe it's useful to mention that this is compiled code.

File: TESTING.md (hunk 1 above)
Comment (lievenvg) [reply TESTING.md]: Removed the heading.

File: TESTING.md (hunk 9 above)
Comment (lievenvg) [reply TESTING.md]: Done.

File: TESTING.md [hunk 10]
```diff
@@ -1,6 +1,151 @@
...
+So the compiled code would look like this:
+
+```html
+<div class="m-accordion" data-qa="id-1234">
+  <div ... class="m-accordion__tab">
+  ...
+</div>
+```
+
+#### Overview
```
Comment (TriangleJuice) [inline TESTING.md]: Sorry, missed this: this heading should be `###`

Comment (lievenvg) [reply TESTING.md]: Done.

**Pull request merged**